crates/lgn-editor-srv/apis/resource_browser.yaml

```diff
@@ -0,0 +1,379 @@
+  version: 1.0.0
+paths:
+  /v1/spaces/{space-id}/workspaces/{workspace-id}/resources/search/:
+    parameters:
+      - $ref: "../../lgn-governance/apis/space.yaml#/components/parameters/space_id"
+      - $ref: "../../lgn-governance/apis/workspace.yaml#/components/parameters/workspace_id"
+      - $ref: "#/components/parameters/token"
+      - $ref: "#/components/parameters/root_resource_id"
+    get:
+      description: Search Resources
+      operationId: SearchResources
+      responses:
+        "200":
+          description: Ok.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/NextSearchToken"
+  /v1/spaces/{space-id}/workspaces/{workspace-id}/resources/:
+    parameters:
+      - $ref: "../../lgn-governance/apis/space.yaml#/components/parameters/space_id"
+      - $ref: "../../lgn-governance/apis/workspace.yaml#/components/parameters/workspace_id"
+    get:
+      description: Get the list of all the resources types available (for creation dialog)
+      operationId: GetResourceTypeNames
+      responses:
+        "200":
+          description: Ok.
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  resource_types:
+                    type: array
+                    items:
+                      type: string
+                required:
+                  - resource_types
+    post:
+      description: Create a resource
+      operationId: CreateResource
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                resource_type:
+                  type: string
+                resource_name:
+                  type: string
+                parent_resource_id:
+                  type: string
+                init_values:
+                  type: array
+                  items:
+                    $ref: "#/components/schemas/InitPropertyValue"
+                upload_id:
+                  type: string
+              required:
+                - resource_type
+                - init_values
+      responses:
+        "200":
+          description: Ok.
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  new_id:
+                    type: string
+                required:
+                  - new_id
+    delete:
+      description: Delete a resource
+      operationId: DeleteResource
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                id:
+                  type: string
+              required:
+                - id
+      responses:
+        "204":
+          description: Ok.
+    put:
```

Comment:
That's a good point that we might have to discuss further with @ereOn . I personally like to use PUT in a truly idempotent way and provide the whole resource to update in the body and use PATCH (with json patch or any other conventions) for partial updates, but maybe we want to stick to the one or the other @ademange-legion  @ereOn ?

For the time being I might keep it as is so that we can update all the put endpoints at once if needed.